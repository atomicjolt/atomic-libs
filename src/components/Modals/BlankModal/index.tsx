import cn from "classnames";
import React from "react";
import { useModal } from "../utils";

export interface BasicModalProps {
  /** Whether or not the modal is visible */
  open?: boolean;
  children?: React.ReactNode;
  /** Centers the modal within the viewport */
  centered?: boolean;
}

/** The most basic Modal that atomic-element offers. Makes no assumptiosn or enforcements on how / where
 * the content of the modal should be rendered.
 *
 * Unless you absolutely need to, you are probably better served using one of the other modals provided
 */
export default function BasicModal(props: BasicModalProps) {
  const { open = false, children, centered = false } = props;
  const renderModal = useModal(open);

  return renderModal(
    <div className={cn("aje-modal-background", { "is-center": centered })}>
      <div className="aje-modal">{children}</div>
    </div>
  );
}
