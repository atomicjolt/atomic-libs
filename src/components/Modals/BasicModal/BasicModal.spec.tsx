import React from "react";
import { describe, test, expect } from "vitest";
import { render } from "@testing-library/react";
import BlankModal from ".";

describe("matches snapshots", () => {
  test("is not visible", () => {
    const result = render(<BlankModal open={false}>Click Me</BlankModal>);
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("is visible", () => {
    const result = render(<BlankModal open={false}>Click Me</BlankModal>);
    expect(result.asFragment()).toMatchSnapshot();
  });
});
