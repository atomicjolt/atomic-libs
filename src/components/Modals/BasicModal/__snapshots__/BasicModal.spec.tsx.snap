// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshots > is not visible 1`] = `<DocumentFragment />`;

exports[`matches snapshots > is visible 1`] = `<DocumentFragment />`;
